### PR TITLE
Fix email validation

### LINKED ISSUE #9554: Update user email validation
### Describe the bug

The email validation fails, when updating a user. Validation on registration works. The bug occured on production.

### Steps to reproduce the behavior

1. Go to  Settings > Users > Edit
2. Change data
2. Click on savve

### Expected behavior

User data can be saved.

### Screenshots

image

### System

- Node.js v14.15.3
- NPM version: 6.14.11
- Strapi version: 3.5.2
- Database: mongo
- Operating system: gcloud run (managed), don't know further

### PR DESCRIPTION
**Pull request by soupette**
Signed-off-by: soupette <cyril.lpz@gmail.com>

### What does it do?

This PR fixes #9554

### CODE REVIEW

File: packages/strapi-admin/admin/src/validations/users/profile.js [hunk 1]
```diff
@@ -10,10 +10,10 @@ const schema = {
     .email(translatedErrors.email)
     .lowercase()
     .required(translatedErrors.required)
-    .test('topLevelDomainValidation', translatedErrors.email, function(value) {
-      let parseDomainResult = parseDomain(value);
-      
-return parseDomainResult.type === ParseResultType.Listed;
+    .test('topLevelDomainValidation', 'Settings.users.validations.email-domain', function(value) {
```
Comment (soupette) [inline packages/strapi-admin/admin/src/validations/users/profile.js:13]: I made the fix but I believe that this is not a wanted logic we want in the admin panel. IMO, this is something that the users should add in their app if they want such validation.

WDYT @alexandrebodin ?

Comment (alexandrebodin) [reply packages/strapi-admin/admin/src/validations/users/profile.js:13]: Not sure what was the wanted behaviour. We can revert it as this is still doable in a lifecycle hook

Comment (soupette) [reply packages/strapi-admin/admin/src/validations/users/profile.js:13]: I think it would be better to revert it yes

Comment (alexandrebodin) [reply packages/strapi-admin/admin/src/validations/users/profile.js:13]: you can go ahead

**Comment by soupette** [comment]
I am closing this PR.

#9588

**Comment by derrickmehaffy** [comment]
This pull request has been mentioned on **Strapi Community Forum**. There might be relevant details there:

https://forum.strapi.io/t/cannot-update-admin-email-on-admin-panel/3285/2

**Pull request closed without merging**